colibri/src/icons.rs

```diff
@@ -119,6 +136,98 @@ async fn query_token_icon_and_extension(
     None
 }
 
+async fn query_uniswap_v3_position_icon(
+    chain_id: u64,
+    token_id: &str,
+    contract_address: Address,
+    inquirer: Arc<EvmNodeInquirer>,
+) -> Option<(Bytes, &'static str)> {
+    if !inquirer.connected_to_any_provider().await {
+        debug!("No connected nodes available for chain ID {} ({})", chain_id, inquirer.blockchain.as_str());
+        return None;
+    }
+
+    let token_id: U256 = match token_id.parse() {
+        Ok(id) => id,
+        Err(e) => {
+            error!("Invalid token ID '{}' for NFT position on chain ID {} ({}): {}", token_id, chain_id, inquirer.blockchain.as_str(), e);
+            return None;
+        }
+    };
+
+    for (node_name, provider) in inquirer.provider_mapping.read().await.iter() {
+        let contract = UniswapV3NFTManager::new(contract_address, provider);
+
+        // try to get the token URI
+        let token_uri = match contract.tokenURI(token_id).call().await {
+            Ok(result) => result._0,
+            Err(e) => {
+                error!("RPC call to tokenURI failed on node '{}' (endpoint: {}) for token ID {} on contract {}: {}",
+                       node_name.name, node_name.endpoint, token_id, contract_address, e);
+                continue;
+            }
+        };
+
+        // process the base64 data from the rpc call
+        let base64_str = match token_uri.strip_prefix("data:application/json;base64,") {
+            Some(data) => data,
+            None => {
+                error!("Invalid token URI format from node '{}' for token ID {} on contract {}: URI does not start with 'data:application/json;base64,'",
+                       node_name.name, token_id, contract_address);
+                break;
+            }
+        };
```

Comment:
```suggestion
        let Some(base64_str) = token_uri.strip_prefix("data:application/json;base64,") else {
            error!("Invalid token URI format from node '{}' for token ID {} on contract {}: URI does not start with 'data:application/json;base64,'",
                node_name.name, token_id, contract_address);
            break;
        };
```

this should also work